viewer/viewer-awt/src/main/java/org/icepdf/ri/images/Images.java

```diff
@@ -15,24 +15,213 @@
  */
 package org.icepdf.ri.images;
 
+import org.icepdf.ri.util.ViewerPropertiesManager;
+
+import javax.swing.*;
 import java.net.URL;
+import java.util.function.Consumer;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.prefs.Preferences;
+
+import static org.icepdf.ri.images.IconPack.Variant;
 
 /**
- * <p>Utility class to allow easy access to image resources
- * Used as an accessor to the images. Just call:</p>
+ * Utility class providing a unified interface to all application icons.
+ * <p/>
+ * For a one-off, directly accessing the default icon pack, call {@link #get(String)}. When loading icons which
+ * are meant to be customizable by applications embedding ICEpdf, call:
  * <ul>
- * <li>Images.get("&lt;filename&gt;.gif")</li>
+ *     <li>for a single icon {@link #getSingleIcon(String, Variant, IconSize)}</li>
+ *     <li>for a button without different styles for its states, {@link #applyIcon(AbstractButton, String, Variant, IconSize)}</li>
+ *     <li>for a button with different styles per state, {@link #applyIcons(AbstractButton, String, IconSize)}</li>
  * </ul>
+ * Icons for the different states will be loaded and registered on the button if the current icon pack provides
+ * these styles. Note that all types of buttons, including {@code JMenu}s and {@code JMenuItem}s are supported
+ * by {@link #applyIcons(AbstractButton, String, IconSize)} and {@link #applyIcon(AbstractButton, String, Variant, IconSize)}.
  *
  * @author Mark Collette
+ * @author Alexander Leithner
  * @since 2.0
  */
 public class Images {
 
+    /**
+     * Enum specifying the wanted size of the icon
+     * <p/>
+     * Note that icon packs may choose to ignore icon sizes
+     */
+    public enum IconSize {
+        /**
+         * "Huge" size, typically aimed to be 48x48 px
+         */
+        HUGE,
+        /**
+         * "Large" size, typically aimed to be 32x32 px
+         */
+        LARGE,
+        /**
+         * "Small" size, typically aimed to be 24x24 px
+         */
+        SMALL,
+        /**
+         * "Mini" size, typically aimed to be 20x20 px
+         */
+        MINI,
+        /**
+         * "Tiny" size, typically aimed to be 16x16 px
+         */
+        TINY
+    }
+
+    private static final Logger LOGGER = Logger.getLogger (Images.class.getName ());
+
+    private static final IconPack ICON_PACK;
+    private static final IconPack.VariantPool AVAILABLE_VARIANTS;
+
+    static {
+        Object defaultIconPackProp = UIManager.get ("org.icepdf.iconpack");
```

Comment:
suggest we name space this the viewer code.   "org.icepdf.ri.iconpack"

Reply:
Sure, will change this.